iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/base_relative_pointer.hpp

```diff
@@ -48,31 +49,49 @@ namespace rp
 /// @note It should be noted that relocating a memory segment will invalidate relative pointers, i.e. relative pointers
 /// are NOT relocatable. This is because the registration mechanism cannot be automatically informed about the copy of a
 /// whole segment, such a segment would have to be registered on its own (and the original segment deregistered).
+
+struct id_t : public cxx::NewType<uint64_t,
+                                  cxx::newtype::DefaultConstructable,
+                                  cxx::newtype::Comparable,
+                                  cxx::newtype::CopyConstructable,
+                                  cxx::newtype::CopyAssignable,
+                                  cxx::newtype::Sortable,
+                                  cxx::newtype::MoveAssignable,
+                                  cxx::newtype::Convertable,
+                                  cxx::newtype::Incrementable,
+                                  cxx::newtype::ConstructByValueCopy,
+                                  cxx::newtype::MoveConstructable>
+{
+    using ThisType::ThisType;
+};
+
+static const id_t NULL_POINTER_ID{std::numeric_limits<uint64_t>::max()};
```

Comment:
I agree with you that `NewType` should be usable as `constexpr`. That's a basic requirement. I think it's definitely do-able, but I would suggest to implement that in a follow-up PR. For me it would be ok to merge this PR here and continue with a follow-up one.

Reply:
`constexpr`-ability will happen as part of #1554